gensim/models/wrappers/fasttext.py

```diff
@@ -270,26 +278,34 @@ def load_model_params(self, file_handle):
         self.wv.max_n = maxn
         self.sample = t
 
-    def load_dict(self, file_handle):
-        (vocab_size, nwords, _) = self.struct_unpack(file_handle, '@3i')
+    def load_dict(self, file_handle, encoding='utf8'):
+        vocab_size, nwords, _ = self.struct_unpack(file_handle, '@3i')
         # Vocab stored by [Dictionary::save](https://github.com/facebookresearch/fastText/blob/master/src/dictionary.cc)
         assert len(self.wv.vocab) == nwords, 'mismatch between vocab sizes'
-        assert len(self.wv.vocab) == vocab_size, 'mismatch between vocab sizes'
-        ntokens, = self.struct_unpack(file_handle, '@q')
+        if len(self.wv.vocab) != vocab_size:
+            logger.warnings("If you are loading any model other than pretrained vector wiki.fr, ")
+            logger.warnings("Please report to gensim or fastText.")
+        ntokens= self.struct_unpack(file_handle, '@1q')
+        if self.new_format:
+            pruneidx_size = self.struct_unpack(file_handle, '@q')
```

Comment:
I don't see those two lines in the code. Do you mean `pruneidx_size` is always zero?
If yes, we should add an assert statement, since if `pruneidx_size` isn't zero and we don't read the corresponding bytes, the subsequent bytes are not going to be parsed correctly.